pkg/kcp/provider/gcp/nuke/providerResourceStatusDeleting.go

```diff
@@ -0,0 +1,38 @@
+package nuke
+
+import (
+	"context"
+	cloudcontrolv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+)
+
+func providerResourceStatusDeleting(ctx context.Context, st composed.State) (error, context.Context) {
+	state := st.(*State)
+
+	changed := false
+
+	for _, prks := range state.ProviderResources {
+		kindStatus, created := state.ObjAsNuke().Status.GetKind(prks.Kind, cloudcontrolv1beta1.ProviderResource)
+		if created {
+			changed = true
+		}
+
+		for _, obj := range prks.Objects {
+			if kindStatus.Objects[obj.GetId()] != cloudcontrolv1beta1.NukeResourceStatusDeleting {
+				kindStatus.Objects[obj.GetId()] = cloudcontrolv1beta1.NukeResourceStatusDeleting
+				changed = true
+			}
+		}
+	}
+
+	if !changed {
+		return nil, ctx
+	}
+
+	state.ObjAsNuke().Status.State = "Deleting"
```

Comment:
Don't patch it if status.state is already "Deleting"
```suggestion
	if state.ObjAsNuke().Status.State != "Deleting" {
		changed = true
		state.ObjAsNuke().Status.State = "Deleting"
	}
	
	if !changed {
		return nil, ctx
	}

```

Reply:
This logic is the exact copy of the kcp/nuke/resourceStatusDeleting.go . That said, I applied your requested change.